services/user.js

```diff
@@ -5,10 +5,24 @@ const { createError } = require('~/utils/errorsHelper')
 const { USER_NOT_FOUND, ALREADY_REGISTERED } = require('~/consts/errors')
 
 const userService = {
-  getUsers: async () => {
-    const users = await User.find().populate('categories').select('-__v').lean().exec()
-
-    return users
+  getUsers: async ({ match, sort, skip, limit }) => {
+    const [{ items, calculations }] = await User.aggregate([
```

Comment:
I have doubts that using aggregate pipeline here is a good idea, you can use it for a separate endpoint like /popular-users, where you filter all users and return popular ones only, lets say top 10, but here filter must be optional, you must return all or filtered result.

Reply:
and why its a bad idea here?

Reply:
answer to this question you can find in my comment itself